src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vilkårsvurdering/VilkårsvurderingUtils.kt

```diff
@@ -253,3 +262,78 @@ fun finnTilOgMedDato(tilOgMed: LocalDate?, vilkårResultater: List<VilkårResult
+                    feiler.add("Vilkår '$vilkårType' for barn med fødselsdato $fødselsdato mangler fom dato.")
+                }
+                if (vilkårResultat.periodeFom != null && vilkårResultat.lagOgValiderPeriodeFraVilkår().fom.isBefore(barn.fødselsdato)) {
+                    feiler.add(
+                        "Vilkår '$vilkårType' for barn med fødselsdato $fødselsdato " +
+                            "har fra-og-med dato før barnets fødselsdato."
+                    )
+                }
+                if (vilkårResultat.periodeFom != null &&
+                    vilkårResultat.erEksplisittAvslagPåSøknad != true &&
+                    vilkårResultat.vilkårType == Vilkår.MELLOM_1_OG_2_ELLER_ADOPTERT
+                ) {
+                    vilkårResultat.validerVilkår_MELLOM_1_OG_2_ELLER_ADOPTERT(
+                        vilkårResultat.lagOgValiderPeriodeFraVilkår(),
+                        barn.fødselsdato
+                    )?.let { feiler.add(it) }
+                }
+            }
+    }
+
+    if (feiler.isNotEmpty()) {
+        throw Feil(feiler.joinToString(separator = "\n"))
+    }
+}
+
+private fun VilkårResultat.lagOgValiderPeriodeFraVilkår(): IkkeNullbarPeriode<Long> = when {
+    periodeFom !== null -> {
+        IkkeNullbarPeriode(verdi = behandlingId, fom = checkNotNull(periodeFom), tom = periodeTom ?: TIDENES_ENDE)
+    }
+    erEksplisittAvslagPåSøknad == true && periodeTom == null -> {
+        IkkeNullbarPeriode(verdi = behandlingId, fom = TIDENES_MORGEN, tom = TIDENES_ENDE)
+    }
+    else -> {
+        throw FunksjonellFeil("Ugyldig periode. Periode må ha t.o.m.-dato eller være et avslag uten datoer.")
+    }
+}
+
+private fun VilkårResultat.validerVilkår_MELLOM_1_OG_2_ELLER_ADOPTERT(
+    periode: IkkeNullbarPeriode<Long>,
+    barnFødselsdato: LocalDate
+): String? = when {
+    this.erAdopsjon() && periode.tom.isAfter(etterAugustBarnetFyller6År(barnFødselsdato.plusYears(6))) ->
+        "Du kan ikke sette en t.o.m dato som er etter august året barnet fyller 6 år."
+    this.erAdopsjon() && periode.tom.diffIDager(periode.tom) > 365 ->
+        "Differansen mellom f.o.m datoen og t.o.m datoen kan ikke være mer enn 1 år."
+    !this.erAdopsjon() && !periode.fom.isEqual(barnFødselsdato.plusYears(1)) ->
+        "F.o.m datoen må være lik barnets 1 års dag."
+    !this.erAdopsjon() && !periode.tom.isEqual(barnFødselsdato.plusYears(2)) ->
+        "T.o.m datoen må være lik barnets 2 års dag."
+    else -> null
+}
+
+fun etterAugustBarnetFyller6År(barnFødselsdato: LocalDate) = when {
+    barnFødselsdato.month > Month.AUGUST -> {
+        barnFødselsdato.plusMonths(((Month.DECEMBER.value - barnFødselsdato.month.value) + Month.AUGUST.value).toLong())
+    }
+    barnFødselsdato.month < Month.AUGUST -> {
+        barnFødselsdato.plusMonths((Month.AUGUST.value - barnFødselsdato.month.value).toLong())
+    }
+    else -> {
+        barnFødselsdato
+    }
+}.førsteDagIInneværendeMåned()
```

Comment:
Hmm, skal man ikke bare sjekke at tom ikke er satt etter august året barnet fyller 6 år?

Med andre ord så kan man gjøre:
`this.erAdopsjon() && periode.tom.isAfter(barnFødselsdato.plusYears(6).withMonth(8).førsteDagIInneværendeMåned())`, også slette hele denne metoden? Kan fort hende jeg misforstår nå da 😊

Reply:
Nei, det kan ikke gjøres. Hvis barnet er født i Oktober 22, bør vi sjekke tom mot 01.08.2029 ikke 01.08.2028.
periode.tom.isAfter(barnFødselsdato.plusYears(6).withMonth(8).førsteDagIInneværendeMåned() kommer til å returnerer alltid 01.08.2028

Reply:
Etter en prat med Eivind:

Barn som er født i Oktober 2022:
```
Ved adopsjon skal dette gjelde:
f.o.m kan aldri være før 1 års dagen til barnet(her tidligst Oktober 2023)
t.o.m kan aldri være lenger enn 1 år frem i tid(hvis f.o.m er oktober 23, er t.o.m maks oktober 24.)
t.o.m kan aldri være lenger enn august året barnet fyller 6 år(Hvis f.o.m er april 2028 vil t.o.m maks kunne være august 2028.)
```

Hvis du tillater 01.08.2029 som det gjøres akkurat nå, så tror jeg at vi slipper igjennom også for august året barnet fyller 7 år.

Reply:
fikset det